src/components/ReportsPage/HeatmapPage/HeatmapFilters.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2023 Cortex Applications, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Checkbox, FormControlLabel, Grid, InputLabel } from "@material-ui/core";
+import React, { ChangeEvent } from "react";
+import { GroupByDropdown } from "../Common/GroupByDropdown";
+import { HeaderTypeDropdown } from "../Common/HeaderTypeDropdown";
+import { HeatmapFiltersModal, ScoreFilters } from "./HeatmapFiltersModal";
+import { GroupByOption, HeaderType, ScorecardLadder } from "../../../api/types";
+import { StringIndexable } from "./HeatmapUtils";
+import { HomepageEntity } from "../../../api/userInsightTypes";
+
+export interface HeatmapPageFilters {
+  selectedScorecardId?: number;
+  groupBy: GroupByOption;
+  headerType: HeaderType;
+  scoreFilters: ScoreFilters;
+  useHierarchy: boolean;
+  hideWithoutChildren: boolean;
+}
+
+interface HeatmapFiltersProps {
+  filters: HeatmapPageFilters;
+  setFiltersAndNavigate: (partialFilters: Partial<HeatmapPageFilters>) => void;
+  entitiesByTag: StringIndexable<HomepageEntity>;
+  excludedGroupBys: GroupByOption[];
+  ladder: ScorecardLadder | undefined;
+}
+
+export const HeatmapFilters: React.FC<HeatmapFiltersProps> = ({ filters, setFiltersAndNavigate, entitiesByTag, excludedGroupBys, ladder }) => {
+  const isHierarchyToggleAllowed = [GroupByOption.TEAM, GroupByOption.DOMAIN].includes(filters.groupBy);
+
+  const onGroupByChange = (event: ChangeEvent<{ value: unknown }>) => {
+    const groupBy = event.target.value as GroupByOption;
```

Comment:
This abstraction is v strange -- we should do this casting in the `<GroupByDropdown>`